core/src/main/java/com/linecorp/armeria/common/HttpHeadersBase.java

```diff
@@ -102,6 +127,53 @@
         endOfStream = parent.isEndOfStream();
     }
 
+    private void copyCachedValues(HttpHeadersBase parent) {
+        method = parent.method;
+        cookies = parent.cookies;
+        acceptLanguages = parent.acceptLanguages;
+        status = parent.status;
+        setCookie = parent.setCookie;
+        contentType = parent.contentType;
+        contentDisposition = parent.contentDisposition;
+    }
+
+    @Override
+    void onChange(@Nullable AsciiString name) {
+        if (isMutating) {
```

Comment:
This sounds somewhat unclear to me. 
How about `!clearCache`? 🤔

Reply:
I'd like to use a positive expression for less confusion. 😀
How about `inMutating` or `preserveCache`?

Reply:
`preserveCache` sounds good to me.